src/vellum/prompts/blocks/compilation.py

```diff
@@ -0,0 +1,190 @@
+                    source=block.chat_source,
+                    blocks=[inner for inner in inner_prompt_blocks if inner.block_type == "VALUE"],
+                    cache_config=block.cache_config,
+                )
+            )
+
+        elif block.block_type == "JINJA":
+            if block.template is None:
+                continue
+
+            rendered_template = render_sandboxed_jinja_template(
+                template=block.template,
+                input_values={input_.key: input_.value for input_ in sanitized_inputs},
+                jinja_custom_filters=DEFAULT_JINJA_CUSTOM_FILTERS,
+                jinja_globals=DEFAULT_JINJA_CUSTOM_FILTERS,
+            )
+            jinja_content = StringVellumValue(value=rendered_template)
+
+            compiled_blocks.append(
+                CompiledValuePromptBlock(
+                    content=jinja_content,
+                    cache_config=block.cache_config,
+                )
+            )
+
+        elif block.block_type == "VARIABLE":
+            compiled_input: Optional[PromptRequestInput] = next(
+                (input_ for input_ in sanitized_inputs if input_.key == str(block.input_variable)), None
+            )
+            if compiled_input is None:
+                raise PromptCompilationError(f"Input variable '{block.input_variable}' not found")
+
+            if compiled_input.type == "CHAT_HISTORY":
+                history = cast(list[ChatMessage], compiled_input.value)
+                chat_message_blocks = _compile_chat_messages_as_prompt_blocks(history)
+                compiled_blocks.extend(chat_message_blocks)
+                continue
+
+            if compiled_input.type == "STRING":
+                compiled_blocks.append(
+                    CompiledValuePromptBlock(
+                        content=StringVellumValue(value=compiled_input.value),
+                        cache_config=block.cache_config,
+                    )
+                )
+            elif compiled_input == "JSON":
+                compiled_blocks.append(
+                    CompiledValuePromptBlock(
+                        content=JsonVellumValue(value=compiled_input.value),
+                        cache_config=block.cache_config,
+                    )
+                )
+            elif compiled_input.type == "CHAT_HISTORY":
+                chat_message_blocks = _compile_chat_messages_as_prompt_blocks(compiled_input.value)
+                compiled_blocks.extend(chat_message_blocks)
+            else:
+                raise ValueError(f"Invalid input type for variable block: {compiled_input.type}")
+
+        elif block.block_type == "RICH_TEXT":
+            value_block = _compile_rich_text_block_as_value_block(block=block, inputs=sanitized_inputs)
+            compiled_blocks.append(value_block)
+
+        elif block.block_type == "FUNCTION_DEFINITION":
+            raise ValueError("Function definitions shouldn't go through compilation process")
```

Comment:
[nit] why not raise `PromptCompilationError` in place of these `ValueError`s?

Reply:
It probably should – I'll update.